packages/block-editor/src/components/link-control/index.js

```diff
@@ -173,7 +184,7 @@ function LinkControl( {
 
 		return Promise.resolve( [
 			{
-				id: '-1',
+				id: uniqueId(),
```

Comment:
@aduth Is this a good idea? Could make testing unreliable. Any better suggestions given that we now have x2 "placeholder" values that aren't from an API that generates IDs.

Reply:
Does it need to be unique? Does it even need to exist?

Based on what I discuss at #19827, https://github.com/WordPress/gutenberg/pull/17846#discussion_r374411692, and https://github.com/WordPress/gutenberg/pull/19856#discussion_r371265123, I might consider instead one of:

- Omit it altogether (or set to `null`) for manually-entered URLs, if it really only makes sense in the context of being associated with an entity within a particular type, and we're okay (based on the above discussions) to diverge the shapes of `value` and `suggestion`
- Or: Set to the URL itself (this is what I proposed in #19827), since even if technically redundant with the `url` property, it's the closest thing to a unique identifier associated with a URL is... the URL itself

Reply:
> Does it need to be unique? Does it even need to exist?

Great questions, and I agree with omitting it altogether. If we set it to a URL, it may not even end up being unique if it's used multiple times, and then... what's the point of the ID if it's not unique?

I've removed it for now and updated snapshots.

Reply:
> > Does it need to be unique? Does it even need to exist?
> 
> Great questions, and I agree with omitting it altogether. If we set it to a URL, it may not even end up being unique if it's used multiple times, and then... what's the point of the ID if it's not unique?

One thing we'll want to be wary of if we do omit it is if we're going to have issues around how we generate a value for `onChange`. Currently, we merge the selected suggestion with the current value to derive the next value. This can be problematic because it may cause properties from the previous value to be unintentionally inherited into the next value.

```js
const currentValue = { id: 1, title: 'My Post', type: 'post', url: 'https://mysite.local/my-post' };
const selectedSuggestion = { type: 'URL', url: 'https://example.com' };
const nextValue = { ...currentValue, ...selectedSuggestion };
// {id: 1, title: "My Post", type: "URL", url: "https://example.com"}
```

Noting that this is the same problem I am seeking to address in #20020. Alternatively, rather than omitting these values, we could choose to set them as explicitly `null`. I think we will want to be consistent with our approach to handling these suggestions.

Reply:
> Currently, we merge the selected suggestion with the current value to derive the next value. This can be problematic because it may cause properties from the previous value to be unintentionally inherited into the next value.

This scenario concerns me. Note also that we're using `suggestion.id` as the React `key` when rendering search results so it needs to be a unique and usable value (ie: not `null`).

I wonder whether we simply create manual "Ids" for these suggestions. We know there will only ever be one of each so we can be sure that they will be unique.

Something like:

```js
{
    id: '1_create_type_id',
    title: 'Some title',
    url: 'Some title',
    type: CREATE_TYPE,
}
```

Here we know that `1_create_type_id` will be unique and also _identifiable_. And for the Direct URL entry type we could have

```
{
    id: '2_url_type_id',
    title: 'Some manually entered URL title',
    url: 'https://someurl.org',
    type: 'url',
}
```

Reply:
>Note also that we're using `suggestion.id` as the React `key` when rendering search results so it needs to be a unique and usable value (ie: not `null`).

It's an implementation detail we should ideally be able to avoid surfacing to consumers in this value shape.

What do you think about the other option I mention in https://github.com/WordPress/gutenberg/pull/19775#discussion_r374799883, which is just to assign the URL as its `id` for manually-entered links?

Reply:
> What do you think about the other option I mention in #19775 (comment), which is just to assign the URL as its id for manually-entered links?

1. What happens if the URL ends up being non-unique? IE, someone entering the same `https://wordpress.org` and `https://wordpress.org` for the IDs? 
2. Also, any concern over security/sanitization of being able to assign whatever a user inputs as the ID? 

I do not have a good understanding of how this all works, just some things that came to mind as potential issues. If it's fine to use the URL, then I'm more than happy to update the PR to match!

Reply:
> 1. What happens if the URL ends up being non-unique? IE, someone entering the same `https://wordpress.org` and `https://wordpress.org` for the IDs?

To me, I would think those _should_ be considered the same (at least conceptually, I'm not sure how this ties in to the implementation).

Reply:
Sounds good to me then! I'll update it to use the URL :)

Reply:
But please can you [check that this doesn't break.](https://github.com/WordPress/gutenberg/pull/19775/files#diff-106913d5079b57c6032c0775292d8d9dR377)